pkg/querier/querier_test.go

```diff
@@ -1220,6 +1220,101 @@ func TestConfig_ValidateLimits(t *testing.T) {
 	}
 }
 
+func TestClampTime(t *testing.T) {
```

Comment:
Could you please add some test cases for scenarios where we'll leave the original time as-is? eg. limit is disabled, original time is already within the desired range